resolvers/product/index.js

```diff
@@ -0,0 +1,55 @@
+const data = require("./productListing.json");
+
+const ProductListingSort = {
+    A_Z: "A_Z",
+    Z_A: "Z_A",
+    LOW_TO_HIGH: "LOW_TO_HIGH",
+    HIGH_TO_LOW: "HIGH_TO_LOW",
+    POPULAR: "POPULAR",
+    NEW: "NEW",
+    FAVOURITES: "FAVOURITES"
+};
+
+function product(_ , { id }) {
+    return data.products.find((product) => product.id === id) || null;
+}
+
+function productSummaryListing(_, { sort }) {
+  let products = data.products;
+
+  switch (sort) {
+    case ProductListingSort.A_Z:
+      products = products.sort((a, b) => (a.name > b.name ? 1 : -1));
+      break;
+    case ProductListingSort.Z_A:
+      products = products.sort((a, b) => (a.name < b.name ? 1 : -1));
+      break;
+    case ProductListingSort.LOW_TO_HIGH:
+      products = products.sort((a, b) =>
+        a.defaultVariant.price.amount.amount >
```

Comment:
I think we should take into consideration sale price also besides `amount`, which is: `was: Money`.

Reply:
Which one should we look up first? IMO, since the `price` is, in theory, the lowest one, it's the best to use as a reference for sorting

Reply:
You are right, I was confused, but yes, `amount` is the final and lowest price.